supabase/migrations/20230411175952_langchain-embedding_search.sql

```diff
@@ -0,0 +1,144 @@
+insert into app.packages(
+    handle,
+    partial_name,
+    control_description,
+    control_relocatable,
+    control_requires
+)
+values (
+    'langchain',
+    'embedding_search',
+    'Search document embeddings',
+    true,
+    '{pg_vector}'
+);
+
+
+
+
+insert into app.package_versions(package_id, version_struct, sql, description_md)
+values (
+(select id from app.packages where package_name = 'langchain-embedding_search'),
+(1,0,0),
+$pkg$
+-- Enforce requirements
+-- Workaround to https://github.com/aws/pg_tle/issues/183
+do $$
+    declare
+        dependencies_exists boolean = exists(
+            select 1
+            from pg_available_extensions
+            where
+                name = 'vector'
+                and installed_version is not null
+        );
+    begin
+
+        if not dependencies_exists then
+            raise
+                exception '"langchain-embedding_search" requires "vector"'
+                using hint = 'Run "create extension vector" and try again';
+        end if;
+    end
+$$;
+
+-- Create a table to store your documents
+create table documents (
+  id bigserial primary key,
+  content text, -- corresponds to Document.pageContent
+  metadata jsonb, -- corresponds to Document.metadata
+  embedding vector(1536) -- 1536 works for OpenAI embeddings, change if needed
+);
+
+-- Create a function to search for documents
+create function match_documents (
+  query_embedding vector(1536),
+  match_count int
+) returns table (
+  id bigint,
+  content text,
+  metadata jsonb,
+  similarity float
+)
+language plpgsql
+as $$
+#variable_conflict use_column
+begin
+  return query
+  select
+    id,
+    content,
+    metadata,
+    1 - (documents.embedding <=> query_embedding) as similarity
+  from documents
+  order by documents.embedding <=> query_embedding
+  limit match_count;
+end;
+$$;
+
+-- Create an index to be used by the search function
+create index on documents
```

Comment:
Lets actually remove this step, I've learned this shouldn't be done on an empty table